tests/php/View/SSViewerTest.php

```diff
@@ -1016,6 +1017,48 @@ public function testIfBlocks()
         );
     }
 
+    public function provideIfBlockWithIterable(): array
+    {
+        $scenarios = [
+            'empty array' => [
+                'iterable' => [],
+                'inScope' => false,
+            ],
+            'array' => [
+                'iterable' => [1, 2, 3],
+                'inScope' => false,
+            ],
+            'iterator' => [
+                'iterable' => new ArrayIterator([1, 2, 3]),
+                'inScope' => false,
+            ],
```

Comment:
I'm surprised this works - but I think it's worth keeping it working, since it doesn't require much additional complexity to maintain.